include/pelib/ImportDirectory.h

```diff
@@ -63,11 +63,19 @@ namespace PeLib
 			{
 				do
 				{
-					FileIter = std::find_if(FileIter, EndIter, std::bind2nd(std::mem_fun_ref(&PELIB_IMAGE_IMPORT_DIRECTORY<bits>::operator==), strFilename));
+					FileIter = std::find_if(
+							FileIter,
+							EndIter,
+							[&](const auto& i) { return i == strFilename; }
+					);
 
 					if (FileIter != EndIter)
 					{
-						typename std::vector<PELIB_THUNK_DATA<bits> >::const_iterator Iter = std::find_if(FileIter->originalfirstthunk.begin(), FileIter->originalfirstthunk.end(), std::bind2nd(std::mem_fun_ref(comp), value));
+						auto Iter = std::find_if(
+								FileIter->originalfirstthunk.begin(),
+								FileIter->originalfirstthunk.end(),
+								std::bind(comp, std::placeholders::_1, value)
```

Comment:
I suggest using lamdas instead of `std::bind()` in C++17 code. For more information, please see _"Item 34: Prefer lambdas to std::bind"_ in [Scott Meyers: Effective Modern C++](https://www.amazon.com/gp/product/1491903996/).

This note concerns several pieces of code in the present PR.

Reply:
In those other cases [where `std::bind()` is used], I was simply not able to replace it with something else. I bet it is possible, but those `ResComparer<T>` are weird and I got lost in some crazy compiler errors.
In this case you commented on, there was actually a lambda originally. I changed it last minute before the commit because I thought `std::bind()` expresses better what is going on. In other cases, there is usually a call to method of the object you get from lambda. Here, this object is used as one of the parameters of `comp` function - so I thought this is expressing that better.
Are reasons for preferring lambdas mentioned in that book applicable to this case?

Reply:
The problem with those comparators is, that it is doing overloading in that single function that uses it - the correct comparator (string/int) is selected based on the type of ID in parameter.

Reply:
In general, lambdas are more readable, more expressive, and may be more efficient than `std::bind` (see the referenced book). In this particular case, I would get rid of that comparators and use overloaded member functions. In my opinion, it makes the code simpler and easier to understand. [Here is a patch](https://files.petrzemek.net/tmp/Replace-comparators-with-overloaded-member-functions.patch) that you can apply with `git am < file.patch` (I was unable to upload it here as GitHub failed with "something went really wrong").

Reply:
Thanks, I applied the patch. It is much better now. I was afraid to touch those comparator structures, but it looks ok in your code.